src/main/java/io/github/fusionflux/portalcubed/content/prop/HammerItem.java

```diff
@@ -0,0 +1,43 @@
+package io.github.fusionflux.portalcubed.content.prop;
+
+import java.util.function.Consumer;
+
+import io.github.fusionflux.portalcubed.data.tags.PortalCubedItemTags;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.storage.loot.LootParams;
+import net.minecraft.world.level.storage.loot.parameters.LootContextParamSets;
+import net.minecraft.world.level.storage.loot.parameters.LootContextParams;
+
+public class HammerItem extends Item {
+	public HammerItem(Properties settings) {
+		super(settings);
+	}
+
+	public static boolean usingHammer(Player player) {
+		return player.getMainHandItem().is(PortalCubedItemTags.WRENCHES);
+	}
+
+	public static void destroyProp(Player user, Level level, Prop prop) {
+		prop.kill();
+		if (level instanceof ServerLevel serverLevel) {
+			if (prop instanceof ButtonActivatedProp buttonActivated)
+				buttonActivated.setActivated(false);
+			consumeEntityLootTable(serverLevel, prop, prop::spawnAtLocation);
+		}
+	}
+
+	private static void consumeEntityLootTable(ServerLevel level, Entity entity, Consumer<ItemStack> lootConsumer) {
```

Comment:
this method (and destroyProp) shouldn't exist. A prop dying should drop its item, with no additional handling.